client/lib/signup/step-actions/index.js

```diff
@@ -411,15 +428,21 @@ function processItemCart(
 
 	const userLoggedIn = isUserLoggedIn( reduxStore.getState() );
 
+	const flowToCheck = stepData.flowName || stepData.lastKnownFlow;
+	const skipSitesAndUsersFetch =
+		isEnabled( 'signup/setup-site-after-checkout' ) &&
+		! newCartItems.length &&
+		[ 'onboarding', 'free', 'business', 'premium', 'personal' ].includes( flowToCheck );
```

Comment:
Checking the a destination might be better?

Reply:
Yeah I'd prefer to check the destination too. I didn't think it would be possible because the flow controller hasn't calculated the destination at the time that the `apiRequestFunction` function is called, but have I missed something?

Reply:
Yep, it's not easy to get the destination here 😅

One way is to make the `_destination` of `SignupFlowController` as independent util function and then we can use it directly
```js
export function getFlowDestination( flowName, isUserLoggedIn, dependencies ) {
	const flow = flows.getFlow( flowName, isUserLoggedIn );
	if ( typeof flow.destination === 'function' ) {
		return flow.destination( dependencies );
	}

	return flow.destination;
}
```